eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -2526,4 +2526,22 @@ public function count(Filter $filter, ?array $languages = null): int
 
         return $this->contentFilteringHandler->count($filter);
     }
+
+    public function loadVersionInfoListByContentInfo(array $contentInfoList): array
+    {
+        $contentIds = array_column($contentInfoList, 'id');
```

Comment:
Moreover you're here relying on magic `id` property which is known to cause performance issue when scaled up.
Instead:
```php
$contentIds = array_map(
    static function (ContentInfo $contentInfo): int {
        return $contentInfo->getId();
    },
    $contentInfoList
);
```